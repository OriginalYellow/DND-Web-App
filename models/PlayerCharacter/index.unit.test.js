/* eslint-disable no-undef */
const {
  tests,
} = require('./index');

test('findBonusSourceIndex should return the default value', () => {
  const bonusTarget = {
    targetName: 'HISTORY',
    targetType: 'SKILL',
    sourceName: 'INT',
    sourceType: 'ABILITY_SCORE',
    value: 10,
    explanation: 'placeholder',
  };

  const bonusSources = [{
    targetName: 'HISTORY',
    targetType: 'SKILL',
    sourceName: 'INT',
    sourceType: 'ABILITY_SCORE',
    value: 10,
    explanation: 'placeholder',
  }];

  expect(tests.findBonusSourceIndex(bonusTarget)(bonusSources))
    .toBe(0);
});

const playerCharacterDocument = {
  abilityScores: {
    wis: {
      name: 'INT',
      value: 17,
      proficient: false,
      bonusSources: [],
      bonusTargets: ['a', 'b'],
    },
    dex: {
      name: 'DEX',
      value: 11,
      proficient: false,
      bonusSources: [],
      bonusTargets: ['c', 'd'],
    },
    str: {
      name: 'STR',
      value: 11,
      proficient: false,
      bonusSources: [],
      bonusTargets: ['e', 'f'],
    },
  },
  skills: {
    acrobatics: {
      name: 'ACROBATICS',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    arcana: {
      name: 'ARCANA',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    athletics: {
      name: 'ATHLETICS',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    history: {
      name: 'HISTORY',
      baseValue: 4,
      proficient: false,
      bonusSources: [{
        targetName: 'HISTORY',
        targetType: 'SKILL',
        sourceName: 'INT',
        sourceType: 'ABILITY_SCORE',
        value: 10,
        explanation: 'placeholder',
      }],
    },
    investigation: {
      name: 'INVESTIGATION',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    nature: {
      name: 'NATURE',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    religion: {
      name: 'RELIGION',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    sleightOfHand: {
      name: 'SLEIGHT_OF_HAND',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
    stealth: {
      name: 'STEALTH',
      baseValue: 4,
      proficient: false,
      bonusSources: [],
    },
  },
};

test('should get the correct bonus targets', () => {
  expect(tests.getBonusTargets(playerCharacterDocument))
    .toEqual(['a', 'b', 'c', 'd', 'e', 'f']);
});

const bonusTargets = [{
  explanation: 'explanation is not yet set',
  sourceName: 'INT',
  sourceType: 'ABILITY_SCORE',
  targetName: 'ARCANA',
  targetType: 'SKILL',
  value: 1,
}, {
  explanation: 'explanation is not yet set',
  sourceName: 'INT',
  sourceType: 'ABILITY_SCORE',
  targetName: 'HISTORY',
  targetType: 'SKILL',
  value: 1,
}, {
  explanation: 'explanation is not yet set',
  sourceName: 'INT',
  sourceType: 'ABILITY_SCORE',
  targetName: 'INVESTIGATION',
  targetType: 'SKILL',
  value: 1,
}, {
  explanation: 'explanation is not yet set',
  sourceName: 'INT',
  sourceType: 'ABILITY_SCORE',
  targetName: 'NATURE',
  targetType: 'SKILL',
  value: 1,
}, {
  explanation: 'explanation is not yet set',
  sourceName: 'INT',
  sourceType: 'ABILITY_SCORE',
  targetName: 'RELIGION',
  targetType: 'SKILL',
  value: 1,
}];

test('should set the bonus sources correctly', () => {
  tests.setBonusSources(bonusTargets, playerCharacterDocument);
  expect(playerCharacterDocument).toMatchSnapshot();
});

test('ability score names should be set', () => {
  tests.setAbilityScoreNames(playerCharacterDocument);
  expect(playerCharacterDocument).toMatchSnapshot();
});
