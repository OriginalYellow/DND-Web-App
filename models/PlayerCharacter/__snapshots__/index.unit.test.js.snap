// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ability score names should be set 1`] = `
Object {
  "abilityScores": Object {
    "dex": Object {
      "bonusSources": Array [],
      "bonusTargets": Array [
        "c",
        "d",
      ],
      "name": "DEX",
      "proficient": false,
      "value": 11,
    },
    "str": Object {
      "bonusSources": Array [],
      "bonusTargets": Array [
        "e",
        "f",
      ],
      "name": "STR",
      "proficient": false,
      "value": 11,
    },
    "wis": Object {
      "bonusSources": Array [],
      "bonusTargets": Array [
        "a",
        "b",
      ],
      "name": "WIS",
      "proficient": false,
      "value": 17,
    },
  },
  "skills": Object {
    "acrobatics": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "ACROBATICS",
      "proficient": false,
    },
    "arcana": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "ARCANA",
      "proficient": false,
    },
    "athletics": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "ATHLETICS",
      "proficient": false,
    },
    "history": Object {
      "baseValue": 4,
      "bonusSources": Array [
        Object {
          "explanation": "placeholder",
          "sourceName": "INT",
          "sourceType": "ABILITY_SCORE",
          "targetName": "HISTORY",
          "targetType": "SKILL",
          "value": 10,
        },
      ],
      "name": "HISTORY",
      "proficient": false,
    },
    "investigation": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "INVESTIGATION",
      "proficient": false,
    },
    "nature": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "NATURE",
      "proficient": false,
    },
    "religion": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "RELIGION",
      "proficient": false,
    },
    "sleightOfHand": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "SLEIGHT_OF_HAND",
      "proficient": false,
    },
    "stealth": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "STEALTH",
      "proficient": false,
    },
  },
}
`;

exports[`should get the correct bonus targets 1`] = `
Object {
  "0": "a",
  "1": "b",
  "2": "c",
  "3": "d",
  "4": "e",
  "5": "f",
}
`;

exports[`should set the bonus sources correctly 1`] = `
Object {
  "abilityScores": Object {
    "dex": Object {
      "bonusSources": Array [],
      "bonusTargets": Array [
        "c",
        "d",
      ],
      "name": "DEX",
      "proficient": false,
      "value": 11,
    },
    "str": Object {
      "bonusSources": Array [],
      "bonusTargets": Array [
        "e",
        "f",
      ],
      "name": "STR",
      "proficient": false,
      "value": 11,
    },
    "wis": Object {
      "bonusSources": Array [],
      "bonusTargets": Array [
        "a",
        "b",
      ],
      "name": "INT",
      "proficient": false,
      "value": 17,
    },
  },
  "skills": Object {
    "acrobatics": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "ACROBATICS",
      "proficient": false,
    },
    "arcana": Object {
      "baseValue": 4,
      "bonusSources": Array [
        Object {
          "explanation": "explanation is not yet set",
          "sourceName": "INT",
          "sourceType": "ABILITY_SCORE",
          "targetName": "ARCANA",
          "targetType": "SKILL",
          "value": 1,
        },
      ],
      "name": "ARCANA",
      "proficient": false,
    },
    "athletics": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "ATHLETICS",
      "proficient": false,
    },
    "history": Object {
      "baseValue": 4,
      "bonusSources": Array [
        Object {
          "explanation": "explanation is not yet set",
          "sourceName": "INT",
          "sourceType": "ABILITY_SCORE",
          "targetName": "HISTORY",
          "targetType": "SKILL",
          "value": 1,
        },
      ],
      "name": "HISTORY",
      "proficient": false,
    },
    "investigation": Object {
      "baseValue": 4,
      "bonusSources": Array [
        Object {
          "explanation": "explanation is not yet set",
          "sourceName": "INT",
          "sourceType": "ABILITY_SCORE",
          "targetName": "INVESTIGATION",
          "targetType": "SKILL",
          "value": 1,
        },
      ],
      "name": "INVESTIGATION",
      "proficient": false,
    },
    "nature": Object {
      "baseValue": 4,
      "bonusSources": Array [
        Object {
          "explanation": "explanation is not yet set",
          "sourceName": "INT",
          "sourceType": "ABILITY_SCORE",
          "targetName": "NATURE",
          "targetType": "SKILL",
          "value": 1,
        },
      ],
      "name": "NATURE",
      "proficient": false,
    },
    "religion": Object {
      "baseValue": 4,
      "bonusSources": Array [
        Object {
          "explanation": "explanation is not yet set",
          "sourceName": "INT",
          "sourceType": "ABILITY_SCORE",
          "targetName": "RELIGION",
          "targetType": "SKILL",
          "value": 1,
        },
      ],
      "name": "RELIGION",
      "proficient": false,
    },
    "sleightOfHand": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "SLEIGHT_OF_HAND",
      "proficient": false,
    },
    "stealth": Object {
      "baseValue": 4,
      "bonusSources": Array [],
      "name": "STEALTH",
      "proficient": false,
    },
  },
}
`;
